src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/forEach/ForeachSourceInspector.java

```diff
@@ -92,60 +113,76 @@ private boolean isEnsuredByPyParentIf(@NotNull ForeachStatement foreach, @NotNul
                 return result;
             }
 
-            private void analyseContainer(@NotNull PsiElement container) {
+            private void analyseContainer(@NotNull final PsiElement container) {
                 final PhpType resolvedType = OpenapiResolveUtil.resolveType((PhpTypedElement) container, container.getProject());
+
                 if (resolvedType == null) {
                     return;
                 }
+
                 final Set<String> types = resolvedType.filterUnknown().getTypes().stream()
-                        .map(Types::getType).collect(Collectors.toSet());
+                                                      .map(Types::getType).collect(Collectors.toSet());
+
                 if (types.isEmpty()) {
                     /* false-positives: pre-defined variables */
                     if (container instanceof Variable) {
-                        final String variableName = ((Variable) container).getName();
+                        final String variableName = ((NavigationItem) container).getName();
+
                         if (ExpressionCostEstimateUtil.predefinedVars.contains(variableName)) {
                             return;
                         }
                     }
+
                     if (REPORT_UNRECOGNIZED_TYPES) {
                         holder.registerProblem(container, patternNotRecognized, ProblemHighlightType.WEAK_WARNING);
                     }
+
                     return;
                 }
 
                 /* false-positives: multiple return types checked only in function/method; no global context */
                 final PsiElement scope = ExpressionSemanticUtil.getBlockScope(container);
-                if (types.size() > 1 && !(scope instanceof Function)) {
+
+                if ((types.size() > 1) && !(scope instanceof Function)) {
```

Comment:
Where the extra line + extra `()` are coming from? 
It would make my reviewer life easier to get the original code style back =)

Reply:
By topics:

1. The extra line is just to make clear the `if()` vs variable initialization. The code is more "spaced" and it turn the readability better in "long term usage".
2. The extra `()` is IDE thing actually. I enabled a lot of Java inspections on past and one of them is something like "unclear expression". It requires that we add an extra `()` in complex conditionals to delimit each part of expression. It is useful in sometimes, but cases like that I don't think that is very useful, but I too don't care about that. But I can undo that if you prefer. Maybe disable this inspection for you project.
3. Oh. Don't care too much with "update code-style" commits, once that it don't modify the code (except if I describe it - actually I merged the `String.format()` part with this commit accidentally, because the idea is that this commit fix only the style and not modify behavior). It is basically to make code readability better and solve all IDE inspections.

Anyway, if you don't like it I could make a new PR with only the real fix.

Reply:
Yes, please (keep the fix and String.format changes only) =)